R/factorialAnalysis.R

```diff
@@ -183,24 +234,71 @@ factorialAnalysis <- function(jaspResults, dataset, options, ...) {
   factorialRegressionANOVA$addColumnInfo(name = "F", title = gettext("F"), type = "number")
   factorialRegressionANOVA$addColumnInfo(name = "p", title = gettext("p"), type = "number")
 
+  if(!is.null(fit))
+    .factorialRegressionANOVAfillTable(factorialRegressionANOVA, options, fit)
+
   jaspResults[["factorialRegressionANOVA"]] <- factorialRegressionANOVA
+
+}
+
+.factorialRegressionANOVAfillTable <- function(factorialRegressionANOVA, options, fit){
+
+  anova <- summary(aov(fit))
+
+  names <- c(names(aov(fit)$coefficients)[-1], "Residuals")
+  anovaFill <- as.data.frame(matrix(0, length(names), 6))
+
+  anovaFill[,1] <- names
+  anovaFill[,2] <- anova[[1]]$Df
+  anovaFill[,3] <- anova[[1]]$`Sum Sq`
+  anovaFill[,4] <- anova[[1]]$`Mean Sq`
+  anovaFill[,5] <- anova[[1]]$`F value`
+  anovaFill[,6] <- anova[[1]]$`Pr(>F)`
+
+  colnames(anovaFill) <- c("terms", "df", "SS", "adjMS", "F", "p")
```

Comment:
just use the constructor directly
```suggestion
  anovaFill <- data.frame(
    terms = names,
    df    = anova[[1]]$Df,
    ... # etc.
  )
```
otherwise, you're coercing everything to character since names is character.

Reply:
✅